share/completions/go.fish

```diff
@@ -36,18 +45,28 @@ complete -c go -n __fish_use_subcommand -a build -d 'compile packages and depend
 
 # clean
 complete -c go -n __fish_use_subcommand -a clean -d 'remove object files'
-complete -c go -n '__fish_seen_subcommand_from clean' -s i -d "remove the corresponding installed archive or binary (what 'go install' would create)"
+complete -c go -n '__fish_seen_subcommand_from clean' -s i -d "remove the corresponding installed archive or binary"
 complete -c go -n '__fish_seen_subcommand_from clean' -s n -d "print the remove commands it would execute, but not run them"
-complete -c go -n '__fish_seen_subcommand_from clean' -s r -d "clean to be applied recursively to all the dependencies of the packages named by the import paths"
+complete -c go -n '__fish_seen_subcommand_from clean' -s r -d "clean to be applied recursively to all the dependencies"
 complete -c go -n '__fish_seen_subcommand_from clean' -s x -d "clean to print remove commands as it executes them"
+complete -c go -n '__fish_seen_subcommand_from clean' -o cache -d 'clean to remove the entire go build cache'
+complete -c go -n '__fish_seen_subcommand_from clean' -o testcache -d 'clean to expire all test results in the go build cache'
+complete -c go -n '__fish_seen_subcommand_from clean' -o modcache -d 'clean to remove the entire module download cache'
 
 # doc
 complete -c go -n __fish_use_subcommand -a doc -d 'run godoc on package sources'
-complete -c go -n '__fish_seen_subcommand_from doc' -s n -d "prints commands that would be executed"
-complete -c go -n '__fish_seen_subcommand_from doc' -s x -d "prints commands as they are executed"
+complete -c go -n '__fish_seen_subcommand_from doc' -o all -d "Show all the documentation for the package"
+complete -c go -n '__fish_seen_subcommand_from doc' -s c -d 'Respect case when matching symbols'
+complete -c go -n '__fish_seen_subcommand_from doc' -o cmd -d 'Treat a command (package main) like a regular package'
+complete -c go -n '__fish_seen_subcommand_from doc' -o short -d 'One-line representation for each symbol'
+complete -c go -n '__fish_seen_subcommand_from doc' -o src -d 'Show the full source code for the symbol'
+complete -c go -n '__fish_seen_subcommand_from doc' -s u -d 'Show documentation for unexported as well as exported'
 
 # env
 complete -c go -n __fish_use_subcommand -a env -d 'print Go environment information'
+complete -c go -n '__fish_seen_subcommand_from env' -o json -d 'prints the environment in JSON format'
+complete -c go -n '__fish_seen_subcommand_from env' -s u -r -d 'unsets NAME to default'
+complete -c go -n '__fish_seen_subcommand_from env' -s w -r -d 'sets NAME=VALUE to environment'
```

Comment:
This reads NAME=VALUE pairs and sets the corresponding environment variables?

"set NAME=VALUE in the environment"?

Reply:
Yep, set NAME=VALUE to GOENV (a file)

Reply:
Okay, after reading through the documentation (why does go not have a manpage btw):

- go uses environment variables like $GOBIN - you could invoke `GOBIN=/bin go something` and it would use that
- If they aren't set, go consults a file in the location $GOENV (default ~/.config/go/env)
- `go env -w` can be used to change the settings in that file

I think this doesn't need to mention "environment" at all, and I have no idea why the go documentation is explaining it in super roundabout ways.

Just "set default values for config variables" should work?

Ideally this would then also complete some of the general purpose variables, or possibly actually use the output of `go env`.